cocotb/share/lib/vhpi/VhpiImpl.cpp

```diff
@@ -551,7 +551,13 @@ GpiObjHdl *VhpiImpl::native_check_create(const std::string &name,
             for (rgn = vhpi_scan(iter); rgn != NULL; rgn = vhpi_scan(iter)) {
                 if (vhpi_get(vhpiKindP, rgn) == vhpiForGenerateK) {
                     std::string rgn_name = vhpi_get_str(vhpiCaseNameP, rgn);
-                    if (rgn_name.compare(0, name.length(), name) == 0) {
+                    std::size_t found = rgn_name.find_first_of("(");
```

Comment:
The original code seems to be incorrect here e.g. if you have two generate labels `FOO` and `FOOBAR`.

Reply:
See #2255, it's a problem in all of the GPI implementations.